monkey/infection_monkey/ransomware/ransomware_payload.py

```diff
@@ -61,7 +61,8 @@ def run_payload(self):
             file_list = self._find_files()
             self._encrypt_files(file_list)
 
-        self._leave_readme()
+        if self._target_dir:
```

Comment:
There's another `if` check on line 95 that also determines whether or not to leave a README. These should both be in the same place. I suggest this:

```python
    def run_payload(self):
        if not self._target_dir:
            return

        LOG.info("Running ransomware payload")

        if self._encryption_enabled:
            file_list = self._find_files()
            self._encrypt_files(file_list)

        if self._readme_enabled:
            self._leave_readme()
```

Then you can remove the first check inside `_leave_readme()` on line 95.